k.makoyid/oop_code/Network.py

```diff
@@ -0,0 +1,40 @@
+import numpy as np
+from Activation_Functions import ActivationFunctions
+
+
+class NeuralNetwork:
+    def __init__(self, input_size, hidden_size, num_classes):
+        self.W1 = np.random.randn(hidden_size, input_size) / np.sqrt(input_size)
+        self.b1 = np.zeros((hidden_size, 1))
+        self.W2 = np.random.randn(num_classes, hidden_size) / np.sqrt(hidden_size)
+        self.b2 = np.zeros((num_classes, 1))
+
+    def forward_propagation(self, X):
+        Z1 = np.dot(self.W1, X) + self.b1
+        A1 = ActivationFunctions.leaky_relu(Z1)

```

Comment:
Set activation function as parameter in constructor (with default value  ActivationFunctions.leaky_relu)